### PR TITLE
1914 engine redesign dask

### PR DESCRIPTION
**Pull request by chukarsten**
PR to add some unit tests for the dask engine to the engine redesign.

### CODE REVIEW

File: evalml/tests/automl_tests/test_dask_engine.py
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_dask_engine.py:114]: Nice. I think this equality check is sufficient here because we check the cv_scores are the same!

File: evalml/tests/automl_tests/test_dask_engine.py
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_dask_engine.py:131]: Yep our submit evaluation job assumes the data is woodwork objects. Since that assumption is not violated during AutoMLSearch and the accessor is coming soon, I think it's fine to not add coverage (and modify the code) to support dataframes!

File: evalml/tests/automl_tests/test_dask_engine.py
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_dask_engine.py:59]: Can you set `n_jobs=1` for these pipelines as well?

**Review by freddyaboulton — CHANGES_REQUESTED** [review]
@chukarsten This is fantastic! The `test_dask_engine` tests are very thorough. I think there are three tests I'd like to see in `test_automl_dask` before merge. The other thing I want to discuss is whether the `dask_testing` util file is actually needed. The rest are minor comments that I don't think would hold up a merge.

File: evalml/tests/automl_tests/test_dask_engine.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_dask_engine.py:217]: Hahaha true, yea that might be force of habit.

File: evalml/tests/automl_tests/test_automl_dask.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_automl_dask.py:43]: This is that iterative algorithm right?? I was about to write an issue for an earning I get about trying to do multitherading withing a spawned process.  Let me play around with this a little.

File: evalml/automl/engine/dask_engine.py
Comment (dsherry) [reply evalml/automl/engine/dask_engine.py:2]: Ah good call @freddyaboulton . Yep I see dask distributed [is a requirement for featuretools](https://github.com/alteryx/featuretools/blob/main/requirements.txt#L7).

As long as `distributed` is getting installed as part of evalml, we can choose whichever representation we want. I'd list it in our requirements if we need a different version from featuretools, but otherwise IMO its fine to omit it in the evalml reqs.

This raises questions for me about why `distributed` is a hard requirement for featuretools rather than an optional one. I don't know the answer. @rwedge @gsheni @thehomebrewnerd any thoughts on that?

There's a future question here too about how hard it is to support different backends like ray--if we can do that well eventually, dask distributed and the other parallel execution frameworks don't need to be core requirements of evalml.

File: Dask Engine Testing.ipynb
Comment (dsherry) [inline Dask Engine Testing.ipynb:1]: @chukarsten please take this notebook off your branch before merging (or make it a docs demo!)

Comment (chukarsten) [reply Dask Engine Testing.ipynb:1]: yup, I need to update it for use with this branch as I think its a design iteration behind, but just wanted to include it for convenience for now in case anyone wanted to mess with it

File: evalml/tests/automl_tests/test_dask_engine.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_dask_engine.py:53]: Good suggestion, @freddyaboulton .  Will do.

File: evalml/tests/automl_tests/test_automl_dask.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_automl_dask.py:54]: I left it there so that anyone reviewing could either reject it or force me to do it lol.

File: evalml/tests/automl_tests/test_automl_dask.py
Comment (freddyaboulton) [reply evalml/tests/automl_tests/test_automl_dask.py:43]: Yea `AutoMLSearch` will pass the`n_jobs` argument to `IterativeAlgorithm` and `IterativeAlgorithm` will include it in the pipeline parameter dicts it generates!

File: evalml/automl/engine/dask_engine.py
Comment (gsheni) [reply evalml/automl/engine/dask_engine.py:2]: This was added as a hard requirement for Featuretools because we add `n_jobs` as as argument to DFS. If a user increased `n_jobs` from 1 (default), we wanted Featuretools to keep working.

Here is the associated line when we added multiprocessing to Featuretools:
- https://github.com/alteryx/featuretools/pull/170/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R11

File: evalml/tests/automl_tests/dask_testing.py
Comment (chukarsten) [reply evalml/tests/automl_tests/dask_testing.py:13]: done

File: evalml/tests/automl_tests/test_automl_dask.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_automl_dask.py:19]: Should be good to go here.

File: evalml/tests/automl_tests/test_automl_dask.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_automl_dask.py:60]: I wasn't so much testing Sequential AutoMLSearch, but explicitly equating the two Parallel and Sequential results

File: evalml/tests/automl_tests/test_automl_dask.py
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_automl_dask.py:24]: Why does the caplog need to be an instance property? Won't that mean that the caplog is shared across tests? I guess I'm worried what would happen if we have two tests running in parallel, one checks that "error happened" is in caplog and the other checks that "error happened" is not in the caplog.

**Review by freddyaboulton — APPROVED** [review]
This is awesome @chukarsten ! 👏

**Review by bchen1116 — APPROVED** [review]
Tests look super solid! Great job with this one. I left a single comment on a typo, but looks good otherwise

File: evalml/tests/automl_tests/test_dask_engine.py
Comment (chukarsten) [reply evalml/tests/automl_tests/test_dask_engine.py:50]: grazi

**Pull request merged**
